test/new-e2e/systemProbe/systemProbeTestEnv.go

```diff
@@ -0,0 +1,135 @@
+	X86_64InstanceIP string
+	StackOutput      auto.UpResult
+}
+
+var (
+	SSHKeyFile           = filepath.Join(".", "/", "aws-ssh-key")
+	vmConfig             = filepath.Join(".", "systemProbe", "config", "vmconfig.json")
+	DD_AGENT_TESTING_DIR = os.Getenv("DD_AGENT_TESTING_DIR")
+	sshKeyX86            = os.Getenv("LibvirtSSHKeyX86")
+	sshKeyArm            = os.Getenv("LibvirtSSHKeyARM")
+)
+
+func NewTestEnv(name, securityGroups, subnets, x86InstanceType, armInstanceType string) (*TestEnv, error) {
+	systemProbeTestEnv := &TestEnv{
+		context: context.Background(),
+		envName: "aws/sandbox",
+		name:    fmt.Sprintf("microvm-scenario-%s", name),
+	}
+
+	awsManager := credentials.NewManager()
+	sshkey, err := awsManager.GetCredential(credentials.AWSSSMStore, "ci.datadog-agent.aws_ec2_kitchen_ssh_key")
+	if err != nil {
+		return nil, err
+	}
+
+	// Write ssh key to file
+	f, err := os.Create(SSHKeyFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	f.WriteString(sshkey)
+
+	stackManager := infra.GetStackManager()
+
+	config := auto.ConfigMap{
+		"ddinfra:aws/defaultARMInstanceType": auto.ConfigValue{Value: armInstanceType},
+		"ddinfra:aws/defaultInstanceType":    auto.ConfigValue{Value: x86InstanceType},
+		"ddinfra:aws/defaultKeyPairName":     auto.ConfigValue{Value: "datadog-agent-kitchen"},
+		"ddinfra:aws/defaultPrivateKeyPath":  auto.ConfigValue{Value: SSHKeyFile},
+		"ddinfra:aws/defaultSecurityGroups":  auto.ConfigValue{Value: securityGroups},
+		"ddinfra:aws/defaultSubnets":         auto.ConfigValue{Value: subnets},
+		"microvm:microVMConfigFile":          auto.ConfigValue{Value: vmConfig},
+		"microvm:libvirtSSHKeyFileX86":       auto.ConfigValue{Value: sshKeyX86},
+		"microvm:libvirtSSHKeyFileArm":       auto.ConfigValue{Value: sshKeyArm},
+	}
+
+	upResult, err := stackManager.GetStack(systemProbeTestEnv.context, systemProbeTestEnv.envName, systemProbeTestEnv.name, config, func(ctx *pulumi.Context) error {
+		awsEnvironment, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+
+		scenarioDone, err := microvms.RunAndReturnInstances(ctx, awsEnvironment)
+		if err != nil {
+			return err
+		}
+
+		for _, instance := range scenarioDone.Instances {
+			remoteRunner, err := command.NewRunner(*awsEnvironment.CommonEnvironment, "remote-runner-"+instance.Arch, instance.Connection, func(r *command.Runner) (*remote.Command, error) {
+				return command.WaitForCloudInit(awsEnvironment.Ctx, r)
+			})
+
+			filemanager := command.NewFileManager(remoteRunner)
+			_, err = filemanager.CopyFile(
+				fmt.Sprintf("%s/site-cookbooks-%s.tar.gz", DD_AGENT_TESTING_DIR, instance.Arch),
+				fmt.Sprintf("/tmp/site-cookbooks-%s.tar.gz", instance.Arch),
+				pulumi.DependsOn(scenarioDone.Dependencies),
+			)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := json.MarshalIndent(upResult.Summary, "", "	")
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Should this be returning an error?

Reply:
It's not really important if the marshaling succeeds. Its only here as a log.